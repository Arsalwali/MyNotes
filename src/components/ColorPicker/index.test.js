import React from 'react';
import { shallow } from 'enzyme';
import ControlledColorPicker from './';

describe('Color picker component', () => {
  it('should render properly', () => {
    const props = {
      color: '#ffffff',
      setColor: jest.fn(),
      onChangeColor: jest.fn(),
    };
    const wrapper = shallow(<ControlledColorPicker {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly with no color props', () => {
    const props = {
      color: null,
      setColor: jest.fn(),
      onChangeColor: jest.fn(),
    };
    const wrapper = shallow(<ControlledColorPicker {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set color if not passed in props', () => {
    const props = {
      color: null,
      setColor: jest.fn(),
      onChangeColor: jest.fn(),
    };
    const wrapper = shallow(<ControlledColorPicker {...props} />);
    expect(props.setColor.mock.calls.length).toBe(1);
  });

});
