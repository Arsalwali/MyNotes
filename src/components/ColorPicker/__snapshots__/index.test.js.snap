// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color picker component should render properly 1`] = `
<ColorPalette
  colors={
    Array [
      "#c0392b",
      "#d81b60",
      "#e74c3c",
      "#5d7a90",
      "#8e44ad",
      "#2980b9",
    ]
  }
  defaultColor={null}
  icon={
    <Text>
      ✔
    </Text>
  }
  onChange={[Function]}
  paletteStyles={Object {}}
  scaleToWindow={false}
  title={null}
  titleStyles={Object {}}
  value="#ffffff"
/>
`;

exports[`Color picker component should render properly with no color props 1`] = `
<ColorPalette
  colors={
    Array [
      "#c0392b",
      "#d81b60",
      "#e74c3c",
      "#5d7a90",
      "#8e44ad",
      "#2980b9",
    ]
  }
  defaultColor={null}
  icon={
    <Text>
      ✔
    </Text>
  }
  onChange={[Function]}
  paletteStyles={Object {}}
  scaleToWindow={false}
  title={null}
  titleStyles={Object {}}
  value={null}
/>
`;
